kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -0,0 +1,157 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.util;
+
+import java.util.Map;
+
+import io.delta.kernel.types.*;
+
+import io.delta.kernel.internal.actions.Metadata;
+
+/**
+ * Utilities related to the column mapping feature.
+ */
+public class ColumnMapping {
+    private ColumnMapping() {}
+
+    public static final String COLUMN_MAPPING_MODE_KEY = "delta.columnMapping.mode";
+    public static final String COLUMN_MAPPING_MODE_NONE = "none";
+    public static final String COLUMN_MAPPING_MODE_NAME = "name";
+    public static final String COLUMN_MAPPING_MODE_ID = "id";
+    public static final String COLUMN_MAPPING_PHYSICAL_NAME_KEY =
+        "delta.columnMapping.physicalName";
+    public static final String COLUMN_MAPPING_ID_KEY = "delta.columnMapping.id";
+
+    /**
+     * Returns the column mapping mode from the given configuration.
+     *
+     * @param  configuration Configuration
+     * @return Column mapping mode. One of ("none", "name", "id")
+     */
+    public static String getColumnMappingMode(Map<String, String> configuration) {
+        return configuration.getOrDefault(
+            COLUMN_MAPPING_MODE_KEY,
+            COLUMN_MAPPING_MODE_NONE);
+    }
+
+    /**
+     * Checks if the given column mapping mode in the given table metadata is supported.
+     *
+     * @param metadata Metadata of the table
+     */
+    public static void checkSupportedColumnMappingMode(Metadata metadata) {
+        String columnMappingMode = getColumnMappingMode(metadata.getConfiguration());
+        switch (columnMappingMode) {
+            case COLUMN_MAPPING_MODE_NONE: // fall through
+            case COLUMN_MAPPING_MODE_NAME:
+                return;
+            case COLUMN_MAPPING_MODE_ID: // id mode is not yet supported
+            default:
+                throw new UnsupportedOperationException(
+                    "Unsupported column mapping mode: " + columnMappingMode);
+        }
+    }
+
+    /**
+     * Helper method that converts the logical schema (requested by the connector) to physical
+     * schema of the data stored in data files.
+     *
+     * @param logicalSchema     Logical schema of the scan
+     * @param physicalSchema    Physical schema of the scan
+     * @param columnMappingMode Column mapping mode
+     */
+    public static StructType convertToPhysicalSchema(
+        StructType logicalSchema,
+        StructType physicalSchema,
+        String columnMappingMode) {
```

Comment:
nit: 8 space indent

Reply:
Fixed in [this](https://github.com/delta-io/delta/pull/2374/files#diff-38491854ca9c7c7860acd852eef7e8477196d0c9690ea55ed1691ada0c8bf631R100) PR which is stcked on this PR. Not changing in this PR to avoid conflicts when rebasing the next PR.